k8s/k8s.go

```diff
@@ -233,15 +235,17 @@ func CmdAddK8s(ctx context.Context, args *skel.CmdArgs, conf types.NetConf, epID
 			logger.Debugf("IPAM result set to: %+v", result)
 		}
 
-		// Create the endpoint object and configure it.
-		endpoint = api.NewWorkloadEndpoint()
-		endpoint.Name = epIDs.WEPName
-		endpoint.Namespace = epIDs.Namespace
+		// Configure the endpoint (creating if required).
+		if endpoint == nil {
```

Comment:
I think we only want to wrap the creation of the wep in this branch, but still set the other spec / metadata fields on the already existing wep.

All-in-all it doesn't really matter since we should only hit this with KDD, and KDD doesn't care about any of this stuff anyway, but I think that would be more correct should we eventually hit this code path with etcd in the future for some reason.

Reply:
not sure I understand that - unless you are saying the existing code is wrong too (irrespective of my libcalico-go change).

Change we chat?

Reply:
I guess my question is "why do we need the if statement here?"

Am I right in thinking that the WEP this code generates will be the same as the wep that already exists, if it does?

And, if the weps don't match, we want to update the existing wep to have the right fields?

Reply:
Ah gotcha.  Could any of those fields have changed?  I'm guessing the ContainerID possibly?

But yeah perhaps I should just have the creation of the new WEP in the if statement and move everything else outside.